### ISSUE TITLE
Show a running tally of expected-value lost per discard, surviving reload

### ISSUE THREAD

**Issue by markafitzgerald1** [markafitzgerald1/cribbage-trainer#19]
Claude Code (Opus 5, High effort): re-scoped 2026-08-18 on Mark's instruction to absorb #24, so this ships as one vertical slice rather than a storage ticket plus a display ticket. Original scope was the tally alone; #24 held the persistence.

## Goal

After each completed discard, show the user how much expected value that choice gave up against the optimal discard, and keep a running average that survives a page reload.

One number, visible in the app, backed by storage that persists between sessions. The point is that a user who plays ten hands today and five tomorrow sees one continuous figure, not a counter that resets when the tab closes.

## Why this is one ticket and not two

#24 was "persist user play stats in browser storage", with no user-visible surface. A storage-only change can only be checked through unit tests or devtools, never by using the app, so it would ship unverified as experienced and defer every real signal to whichever ticket finally displayed it — with the schema mismatch surfacing there rather than here. Folding it in makes the increment testable the way a user tests it: play a few hands, refresh, confirm the number is right and still there.

## Scope

- Compute EV loss for each completed discard: the chosen keep's expected value minus the best available, using the existing analysis. Nothing new is derived — the analysis table already computes both halves.
- Persist the accumulated tally (count and running mean, at minimum) in browser storage, versioned so the shape can evolve without discarding history silently.
- Display the running average, plus enough context to read it — at least the number of decisions behind it.
- Work with analytics consent declined. This is local data and must not depend on telemetry.

## Not in scope

- The trend over time and any charting: #719.
- Comparison against other users: #666, which needs #665's cross-user data.
- Play (pegging) decisions: only discards have a ground-truth expected value today.

## Notes

- The original body noted this needs the "New Deal" button of #6, which now exists as the Deal button.
- Storage schema decisions made here are what #719 builds on, so prefer a shape that can carry per-decision records rather than only aggregates — a running mean alone cannot produce a trend later.

**Comment by markafitzgerald1** [comment]
Once #15 is done I believe that this provides an better training experience. Without #15 this doesn't have as much value (on its own).

**Comment by markafitzgerald1** [comment]
Perhaps after this Issue has been resolved additional summary statistics could be calculated, such as % of discards that were optimal, different percentiles discard points below optimal and common sources of user error (e.g. tends to miss flush opportunity, tends to miss 15s that are more valuable than runs or vice versa).

**Comment by markafitzgerald1** [comment]
Claude Code (Opus 5, High effort): design note from a roadmap discussion, recording the rule that makes this measurable without depending on #14.

## Score the first complete discard per hand, and only that

Measuring discard quality does **not** require turning auto-analyze off (#14). Analysis fires only once two cards are committed, so with auto-analyze on the user's first complete discard for a hand is already made before any ranked answer for that hand is visible. That first choice is uncontaminated by construction.

What exposure contaminates is a *revision* of the same hand: once the answer key is on screen, changing the discard is an informed choice and says nothing about unaided skill. So the local metric must:

- key on the hand, counting only the **first** complete discard for it;
- exclude, or record separately, any later discard on the same hand;
- exclude hands whose answers were revealed before the user chose at all — a deep link or a history hydration lands on a complete discard with the full ranking already visible, so nothing about that hand is first instinct.

## The vocabulary already exists

`src/ui-react/useDiscardTelemetry.ts` already implements this discipline for GA4: a per-hand `deal_nonce`, a 1-based `analysis_index`, a `source` of `interactive`/`deeplink`/`history`, and `is_first_analysis`, which is true only when the source is interactive *and* the index is 1. #665 keys its expected-value-loss metric off that flag for exactly this reason.

The local metric needs the same rule, implemented against browser storage rather than analytics, because these stats must work with analytics consent declined. Worth reusing the concept and the naming rather than inventing a parallel one — and worth checking whether the nonce and first-exposure logic can be shared rather than duplicated, since `jscpd` runs at 0%.

## Consequence for the metric's meaning

The denominator is *hands where the user chose unaided*, not *hands seen*. That should be surfaced somewhere, because a user who deep-links or replays a lot will otherwise wonder why their counted hands lag their played hands. It also means the metric stays honest if #14 is later enabled: the same rule holds either way, so history does not break when the setting changes.

**Comment by markafitzgerald1** [comment]
Claude Code (Opus 5, High effort), on Mark's behalf: plan for #19, stacked on #732 so it inherits `getDiscardQuality`.

## What already exists

`src/analysis/discardQuality.ts` was written in #732 to be shared with this issue, and it is already computed at the moment this ticket needs. `ScoredPossibleKeepDiscards` derives `quality` into `renderedAnalysis` and fires `onAnalysisRendered` once the scoring tables are on screen. So the EV-loss half of #19 is done and reviewed; what remains is identity, persistence, and display.

## The hazard the issue does not name

`onAnalysisRendered` fires on **every** render where results are on screen, not once per decision. Back and Forward re-show a completed discard while the component stays mounted, and an analysis-table sort change re-renders the same decision. A tally that counts that callback counts one decision three or four times, and the error is invisible: the average simply drifts low or high with navigation habits, with nothing on screen disagreeing.

#665 solves the same problem with exposure bookkeeping keyed by the discard plus a per-hand `dealNonce`, but that state lives inside `useDiscardTelemetry`, which is consent-gated. This issue must work with analytics declined, so it cannot reuse it and must not import it. It needs its own per-deal identity.

Identity will therefore be `(deal occasion, discard)`: the same two cards discarded from a genuinely new deal counts twice, because it is two decisions; the same decision revisited by Back, Forward, or a re-sort counts once. That is a behavioral claim worth challenging in review, and it will be asserted by test rather than left implicit.

## Storage shape

Per-decision records, versioned, as decided:

- A `version` field, so the shape can change without silently discarding history.
- Per-decision records carrying at least the loss, the optimal flag, the crib role, and when it happened — enough for #719 to draw a trend and to segment, which a running mean alone can never reconstruct.
- **Plus** lifetime aggregate counters, deliberately redundant. Per-decision records cannot grow forever in `localStorage`, so records will be bounded; if the bound is ever reached and old records are dropped, a mean computed only from surviving records would silently change. Keeping counters means trimming history costs the trend detail but never the headline number.

Reads fail soft in the manner `readAnalyticsChoice` already establishes here: malformed or absent storage yields an empty tally rather than throwing, and a write that fails — quota, disabled storage, private browsing — leaves the app working without a tally rather than breaking the deal.

## Open questions I intend to decide rather than block on

- **Do seeded, deep-linked, and manually entered hands count?** #665 segments them out of population statistics as practice rather than authentic play. The same argument applies to a personal average: a hand typed in to study a known position is not a decision made under uncertainty, and letting it move the number makes the number mean less. I plan to record the provenance on each decision and exclude non-authentic hands from the headline average, keeping the records so a later view can show them separately. This is the design decision most worth arguing with.
- **Where it renders.** Any new visible element in the side-by-side left column risks the `min-content` invariant that keeps the controls narrower than the six cards, and any rem floor risks the portrait overflow trap. Placement will be chosen against those guards, with an e2e assertion rather than a screenshot alone.

## Not in scope

The trend chart (#719), cross-user comparison (#666), and pegging decisions, per the issue.

## Delivery

Branch `feature/19-discard-ev-tally` off #732's branch, PR opened as a draft as soon as it has one commit, Copilot requested, Codex loop run to a clean round, and a review guide plus manual testing plan in the PR body before any human is asked to look.

**Comment by markafitzgerald1** [comment]
Claude Code (Sonnet 5, High effort): Shipped in #738 (merged 50ed909, squashed to main), stacked on #732 which delivered getDiscardQuality. Per-discard EV loss is computed from the existing analysis, persisted in a versioned localStorage tally (count and running mean, with per-decision records so #719 has something to build a trend from), displayed as "Lost per discard" alongside "Best choice" and "Hands skipped," and works with analytics consent declined since it's local-only data.

One thing worth flagging for anyone reading this later: getting decision identity right took more than the original scope anticipated. The store's dedup is by hand (cards + crib role), which correctly treats a re-sort, Back/Forward, or reload of the same hand as one decision — but a second, narrower identity (telemetry's own per-hand handId) turned out to be necessary too, to tell apart two genuinely different occurrences of identical cards (e.g. a practiced hand later coinciding with a real deal). #738's PR body has the full writeup under "Two identities, not one."
